src/backend/distributed/operations/create_shards.c

```diff
@@ -217,9 +217,13 @@ CreateColocatedShards(Oid targetRelationId, Oid sourceRelationId, bool
 	List *insertedShardPlacements = NIL;
 	List *insertedShardIds = NIL;
 
-	/* make sure that tables are hash partitioned */
-	CheckHashPartitionedTable(targetRelationId);
-	CheckHashPartitionedTable(sourceRelationId);
+	CitusTableCacheEntry *targetCacheEntry = GetCitusTableCacheEntry(targetRelationId);
+	if (targetCacheEntry->partitionMethod != DISTRIBUTE_BY_NONE)
+	{
+		/* make sure that tables are hash partitioned */
+		CheckHashPartitionedTable(targetRelationId);
+		CheckHashPartitionedTable(sourceRelationId);
+	}
```

Comment:
even more, we can drop this check because at some we anyway check whether two tables can be colocated (in ColocationIdForNewTable())

Reply:
I suppose range-partitioned tables can be co-located

Reply:
so do you think we already shouldn't have had those checks?

Reply:
maybe change to an assert

Reply:
On main, this is what happens if try using colocate_with option for range distributed tables:
```sql
SELECT create_distributed_table('tbl_1', 'a', distribution_type=>'range', colocate_with=>'tbl');
ERROR:  cannot distribute relation
DETAIL:  Currently, colocate_with option is only supported for hash distributed tables.
```

So I think removing those checks / converting them to Assert()s would mean a behavioral change that this PR is not expected to do, what do you thin?

Reply:
that's not this error though, can we really get here? it does not appear in tests

Reply:
I think those are more like sanity checks, so let me simply make them asserts then.

Reply:
Oh we can't make them asserts because their return type is `void`, so let me leave them as is.

Reply:
```
Assert(targetCacheEntry->partitionMethod == DISTRIBUTE_BY_HASH ||
       targetCacheEntry->partitionMethod == DISTRIBUTE_BY_NONE);